components/post_view/post_message_view/post_message_view.tsx

```diff
@@ -13,63 +13,35 @@ import PostMarkdown from 'components/post_markdown';
 import Pluggable from 'plugins/pluggable';
 import ShowMore from 'components/post_view/show_more';
 
-export default class PostMessageView extends React.PureComponent {
-    static propTypes = {
-
-        /*
-         * The post to render the message for
-         */
-        post: PropTypes.object.isRequired,
-
-        /*
-         * Set to enable Markdown formatting
-         */
-        enableFormatting: PropTypes.bool,
-
-        /*
-         * Options specific to text formatting
-         */
-        options: PropTypes.object,
-
-        /**
-         * Set to render post body compactly
-         */
-        compactDisplay: PropTypes.bool,
-
-        /**
-         * Flags if the post_message_view is for the RHS (Reply).
-         */
-        isRHS: PropTypes.bool,
-
-        /**
-         * Whether or not the RHS is visible
-         */
-        isRHSOpen: PropTypes.bool,
-
-        /**
-         * Whether or not the RHS is expanded
-         */
-        isRHSExpanded: PropTypes.bool,
-
-        /*
-         * Logged in user's theme
-         */
-        theme: PropTypes.object.isRequired,
-
-        /*
-         * Post type components from plugins
-         */
-        pluginPostTypes: PropTypes.object,
-        currentRelativeTeamUrl: PropTypes.string.isRequired,
-    };
+type Props = {
+    post: Post; /* The post to render the message for */
+    enableFormatting?: boolean; /* Set to enable Markdown formatting */
+    options?: any; /* Options specific to text formatting */
+    compactDisplay?: boolean; /* Set to render post body compactly */
+    isRHS?: boolean; /* Flags if the post_message_view is for the RHS (Reply). */
+    isRHSOpen?: boolean; /* Whether or not the RHS is visible */
+    isRHSExpanded?: boolean; /* Whether or not the RHS is expanded */
+    theme: any; /* Logged in user's theme */
+    pluginPostTypes?: any; /* Post type components from plugins */
```

Comment:
The type of tihs is already defined in `types/store/plugins.ts` in the mattermost-webapp repo. You can pick that same types and use it here, I think it would be `{[postType: string]: PostPluginComponent}`.

Reply:
I need some advice on this, the `PostPluginComponent` type has the `component` property witch is not of a correct type (`React.Component` instead of `React.ComponentType`) so I cannot use it, should I make a PR to the `mattermost-redux` repo with a fix?

Reply:
Would be great but for now, lets keep it here as any.